### ISSUE TITLE
[BUG] Call HTTP in mobile view

### ISSUE THREAD

**Issue by Tosko4** [spotweb/spotweb#525]
Hi there,

I’ve searched the closed issues to see if this has been requested already, couldn’t find it. But please let me know if I was unable to find it. 

My feature request: I am using NZBget over http. Got it all configured propperly and from regular spotweb view, it works fine sending the nzb file over to nzbget. But when I browse with my phone on spotweb, the mobile view used. When I press the download button there, my browser asks me where I want to save the nzb file. 

Expected results should be that the mobile view should handle the download the same way. Call nzbget over http. 

Thanks in advance!

**Comment by Sweepr** [comment]
@Tosko4, mesa57 just recently fixed alot of things concering the mobile template as this has been broken for quite some time, i'm not sure if NZBget support has fully been implemented yet.

@mesa57, can you confirm this?

**Comment by mesa57** [comment]
I tested with NZBGET and that worked too.

**Comment by Tosko4** [comment]
Let me show you please, 

https://drive.google.com/file/d/13sx5QVkBUWK4g7Oa8s_cbuOat0vLcZ7I

**Comment by mesa57** [comment]
Wow nice video 😄 
As you took the effort of making this, I will test again. 👍

**Comment by mesa57** [comment]
Still works here. You should see this button when you open a spot : 

![afbeelding](https://user-images.githubusercontent.com/6350469/79685579-162d4100-823a-11ea-8a9e-50b09818d25b.png)

My nzbget configuration in spotweb : 

![afbeelding](https://user-images.githubusercontent.com/6350469/79685619-71f7ca00-823a-11ea-9656-c84d57f2b413.png)

**Comment by mesa57** [comment]
Also look in the error log if you see something like this : 
`[19-Apr-2020 12:45:26 Europe/Amsterdam] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Failed to connect to nas5 port 6789: Connection refused`

**Comment by Tosko4** [comment]
Hi,my settings:
![8789A667-3D45-4CEF-96F9-EA8FB062A7A4](https://user-images.githubusercontent.com/1294707/79691026-a16bfe00-825d-11ea-94cc-938a6d2810c9.jpeg)

I checked my docker logs. No error, any other place where these things are logged on spotweb itself?

Besides, i dont think it’s a connection issue, as the desktop variant works fine.. so connection is there and it works. But for some reason my mobile template does not recognize it?

**Comment by mesa57** [comment]
The log is the standard php log of you're server (see php.net/error_log)
Btw. why do you use a docker IP address instead of exposing the 6789 port to the host and use the host IP address ? Docker IP addresses can change ....

**Comment by Tosko4** [comment]
I have created my own network subnet inside docker. Regular bridge network is 172.17.0.0/16

I created 172.18.0.0/16 network and allowed traffic from and to this network from my lan (192.158.1.*). This way I can set a static IP to a container. Even after reboot if keeps that IP :)

I could also put the host IP with port there. Would work too 192.168.1.18:6789

I’ll check the logs and will get back

**Comment by mesa57** [comment]
I also use docker (for NZBGET) and have same bridge network 172.17.0.0/16
But I observed sometime a docker IP changes. It is not necessary to route from docker to you're host.
You either specify the virtual port to the host or do host networking on you're container.

**Comment by Tosko4** [comment]
@mesa57 the default bridge 172.17.0.0/16 cannot assign static IP’s. Thats why you experience that. 

Create your own and you can.

**Comment by Tosko4** [comment]
@mesa57 

The only errors i find were on april 18th, when the API was not configured properly yet by me.
Since then, until today no errors or whatsoever.

I just went to the mobile page again, clicked on the button NZB, as expected starts to download the NZB file. Reloaded the logs, no errors at all.

I believe the mobile theme is not recognising it for some reason. It works on the desktop version, so the API is fine, it's more a frontend issue rather than a backend issue.

If you'd like, I can set up a teamviewer session for you so you can have a look yourself?

```
[Sat Apr 18 13:16:36.128068 2020] [php7:notice] [pid 55] [client 192.168.1.18:43442] The configured NZB handler has no api support, referer: http://192.168.1.18/
[Sat Apr 18 13:16:37.038851 2020] [php7:notice] [pid 55] [client 192.168.1.18:43442] NZBGet RPC: Method 'status', Access denied (401), referer: http://192.168.1.18/?page=render&tplname=usermanagement
[Sat Apr 18 13:16:37.146903 2020] [php7:notice] [pid 55] [client 192.168.1.18:43442] NZBGet RPC: Method 'status', Access denied (401), referer: http://192.168.1.18/
```

**Comment by mesa57** [comment]
I think you're error-log is not configured otherwise you would see an error like i had.
The error is generated on start of spotweb.
I am not going to review you're configuration as i am not a specialist in that.
I can only tell you how things are configured on my (working) side.
Btw. you could also test the mobile template in you're local browser.
Firefox has a ctrl+shift+M mode to simulate a phone.

**Comment by Tosko4** [comment]
@mesa57 did that too just now. You were just a little faster than me.
Maybe this helps.. I wasn't signed in to spotweb and went to mobile view on my desktop. Weird thing is.. it shows "logout" button. When I click on that a deprecated error shows. 

I believe my mobile page is not using my credentials. The only account currenly allowed to use the NZBget api is my own, which I need to be signed into in order for that to work. Doesn't matter if i log in on the desktop version first, and then switch to mobile version. It seems like it's not using the same log in session and see's me as logged out from mobile.

Check out this vid:

https://drive.google.com/open?id=1huJn5zkFNkn7HxBwJGFPfU67jptnOzVh

@Sweepr too?

**Comment by mesa57** [comment]
Could be. Normally when you are not logged in the mobile template is not useable and you should get the standard template on you're phone. Only after login the mobile template should become visible.

**Comment by Tosko4** [comment]
Ok so behaviour is as following:

- When I sign out, clear cache en go to the spotweb website, it shows me the regular desktop website. Which is fine.
- Then I switch to the mobile emulator in chrome
- It shows me the mobile template with the logout button etc. But can't be signed in because I was signed out.
- when I go back to the desktop view, login and go back to the mobile emulator in chrome, it remains on the desktop page. Weird...

But this is probably the reason why I don't see the pull nzb button because my mobile theme does nothing with the logged in sessions or whatsoever @Sweepr @mesa57

**Comment by Tosko4** [comment]
Here is the vid of everything I do, showing you the settings. Showing you how NZBget is configured and the behaviour as described above:

https://drive.google.com/open?id=1huJn5zkFNkn7HxBwJGFPfU67jptnOzVh

**Comment by mesa57** [comment]
I have no idea how you reach the mobile template.
It is not configured in you're smartlab user settings.

**Comment by mesa57** [comment]
Btw. I see you defined you're own permissions group (use downloads manager API) which is not needed as trusted user already have that authority.
Nevertheless I tried to add such a group, but it did not make any difference for me. Still works as expected. So in my opinion this is NOT a confirmed bug.

**Comment by mesa57** [comment]
And also, did you follow these steps ?
https://github.com/spotweb/spotweb/wiki/How-to-enable-the-mobile-template.

**Comment by mesa57** [comment]
I also managed to reproduce the warning : 

`jquery.min.js:2 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.`

This happens when you are in mobile, not logged in but click on the logout button.
Apparently in you're situation you are not logged in. Hence the push nzb to nzbget is not working.

**Comment by Tosko4** [comment]
> Btw. I see you defined you're own permissions group (use downloads manager API) which is not needed as trusted user already have that authority.
> Nevertheless I tried to add such a group, but it did not make any difference for me. Still works as expected. So in my opinion this is NOT a confirmed bug.

This was something in the wiki. That’s why i have implemented this. Then we should remove it there

> And also, did you follow these steps ?
> https://github.com/spotweb/spotweb/wiki/How-to-enable-the-mobile-template.

No i did not follow these steps as the mobile view was showing, assuming there was nothing more for me to do. Did you do this yourself?

**Comment by mesa57** [comment]
Yes, I did. Default non authenticated users have no access to the mobile template.
The smartlab user also has no access i saw in the video.
You must set the template to use in user preferences default for all 3 is we1rdo and not mobile.

**Comment by mesa57** [comment]
Problem solved ?

**Comment by Tosko4** [comment]
Ok so I have implemented the change you mentioned, i changed mobile and tablet view to use template mobile.

But the problem remains... When i clear cache etc, there is no logged in session. I am not able to sign in from the mobile view template because the "logout" button remains. I would like to see a login button, to enter my credentials.

Any other clue?

**Comment by Tosko4** [comment]
Managed to solve the problem with a workaround. I went into the code and saw how php pages work.

So i went to <my ip>/index.php?page=login

I got to see a login page, when i enter my credentials and go to a spot, et voila! Push NSB button appears!

So I guess we only have to fix the fact that the mobile view always shows a logout button whether you're logged in or not. To have it check it's state and if not logged in, show you the login page.
